src/js/actions/transform.js

```diff
@@ -1124,9 +1124,18 @@ define(function (require, exports) {
             this.dispatch(events.ui.TOGGLE_OVERLAYS, { enabled: true });
 
             var appStore = this.flux.store("application"),
-                currentDoc = appStore.getCurrentDocument();
+                currentDoc = appStore.getCurrentDocument(),
+                textLayers = currentDoc.layers.allSelected.filter(function (layer) {
+                    // Reset these layers completely because their impliedFontSize may have changed
+                    return layer.kind === layer.layerKinds.TEXT;
+                }),
+                otherLayers = currentDoc.layers.allSelected.filterNot(function (layer) {
+                    return layer.kind === layer.layerKinds.TEXT;
+                }),
+                textLayersPromise = this.flux.actions.layers.resetLayers(currentDoc, textLayers),
+                otherLayersPromise = this.flux.actions.layers.resetBounds(currentDoc, otherLayers);
```

Comment:
These sorts of checks ought to happen exactly once, as deep as possible, so that clients do not constantly have to check for this condition. (I think this is the right way to do it because 1 check has many fewer lines of potentially buggy code than N different checks.) 

Both `resetLayers` and `resetBounds` correctly handle the case when they are passed empty lists, but they don't drop out early. I've pushed commit 7bedadc which additionally causes those actions to resolve immediately when they're applied to empty lists. So I'm now satisfied that it is correct to omit the check here. Good point of discussion.